lib/dao.js

```diff
@@ -1588,38 +1589,15 @@ DataAccessObject.find = function find(query, options, cb) {
         });
       });
 
-      // FIXME: apply "includes" and other transforms - see allCb below
       memory.all(modelName, geoQueryObject, options, allCb);
     } else {
       cb(null, []);
     }
   }
 
-  function geoCallbackWithNotify(err, data) {
```

Comment:
I'm wondering what was the original intent for `geoCallbackWithNotify` and why we are removing it now?

Reply:
e.g. we have it in [3.x](https://github.com/strongloop/loopback-datasource-juggler/blob/3.x/lib/dao.js#L1596-L1619) branch, so maybe we need to revert changes introduced by #1690?

Reply:
In https://github.com/strongloop/loopback/pull/4194, you changed the way how query results are handled when the geo query is handled in-memory.

Before your changes, `geoCallbackWithNotify` was notifying "loaded" observers and calling `cb` directly.

After your change, `geoCallbackWithNotify` was calling `allCb` that notifies "loaded" observers, but at the same time it was  notifying "loaded" observers too, which caused double notifications.

My change is fixing the problem by removing `geoCallbackWithNotify` and always relying on `allCb` to notify "loaded" observers.

I believe the 3.x branch is not affected, because it still calls `cb` instead of `allCb`, see here:
https://github.com/strongloop/loopback-datasource-juggler/blob/548f3018a92b17e2e206971bcfdf541191a1410e/lib/dao.js#L1590

Reply:
Ok thanks for the clarification @bajtos. Do you intend to backport this change into `3.x` so we have consistent behaviour too?

Reply:
I am thinking about back-porting the following two commits:
 - Test "near" queries executed in memory
 - Fix "access" hook for unoptimized "near" queries
 - Make any other adjustments to make the tests pass

I believe that the commit _Fix "loaded" hook for unoptimized "near" queries_ is required only because of  #1690. AFAICT, #1690 was not back-ported to 3.x. WDYT, should we try to back-port it together with my fixes?

Reply:
On the second though, I am also going to rename `geoCallbackWithoutNotify` to `geoCallback` to avoid possible confusion. Even though `geoCallback` does not notify hook observers directly, it calls `allCb` which does trigger notifications.

Reply:
Backport: https://github.com/strongloop/loopback-datasource-juggler/pull/1742

Reply:
oh I see, in that case those two commits should be fine 👍. Agreed with the rename too.